### PR TITLE
Java Lang Server Prototype

### PR DESCRIPTION
**Pull request by etsai-stripe**
### Motivations
Build java lang server prototype to establish java api hover support. this is only a prototyping PR. will have a followup PR to add tests and clean up.
https://jira.corp.stripe.com/browse/DX-6795

### Prototype run recording:
https://user-images.githubusercontent.com/88805634/137003067-0d45f7d5-f0a3-433f-953a-f2553c41765c.mp4

### Hover screenshot
Screen Shot 2021-10-12 at 11 15 21 AM

### Stripe Language Client Output
Screen Shot 2021-10-12 at 1 16 09 PM

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
       "outFiles": ["${workspaceFolder}/dist/**/*.js"],
       "preLaunchTask": "Build all",
       "env": {
-        "EXTENSION_MODE": "development"
+        "EXTENSION_MODE": "development",
+        "DEBUG_VSCODE_JAVA":"true"
```
Comment (gracegoo-stripe) [inline .vscode/launch.json:27]: What does the DEBUG_VSCODE_JAVA do?

File: src/stripeJavaLanguageClient/utils.ts [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+    getJavaConfiguration().get<boolean>('import.maven.enabled') || false;
+  const isGradleEnabled: boolean =
+    getJavaConfiguration().get<boolean>('import.gradle.enabled') || false;
+  if (isMavenEnabled && isGradleEnabled) {
+    const activeBuildTool: string | undefined = context.workspaceState.get(ACTIVE_BUILD_TOOL_STATE);
+    if (!activeBuildTool) {
+      if (!(await hasBuildToolConflicts())) {
+        return true;
+      }
+      outputChannel.appendLine('Build tool conflicts are detected in workspace.');
```
Comment (gracegoo-stripe) [inline src/stripeJavaLanguageClient/utils.ts:105]: What should the users do in this case?

File: src/stripeJavaLanguageClient/utils.ts [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+  });
+}
+
+async function getBuildFilesInWorkspace(): Promise<Uri[]> {
+  const buildFiles: Uri[] = [];
+  const inclusionFilePatterns: string[] = getBuildFilePatterns();
+  inclusionFilePatterns.push('**/.project');
+  const inclusionFolderPatterns: string[] = getInclusionPatternsFromNegatedExclusion();
+  // Since VS Code API does not support put negated exclusion pattern in findFiles(),
+  // here we first parse the negated exclusion to inclusion and do the search.
```
Comment (gracegoo-stripe) [inline src/stripeJavaLanguageClient/utils.ts:145]: When is this used? Is this for cases where we want to exclude everything in `/path/a` unless it's `/path/a/b`?

Comment (etsai-stripe) [reply src/stripeJavaLanguageClient/utils.ts:145]: this it to handle the java configuration settings. ie:
```
"java.import.exclusions": [
     "**/node_modules/**",
    "**/.metadata/**",
    "**/archetype-resources/**",
    "**/META-INF/maven/**",
    "**/Frontend/**",
],
```
if user puts `!` in front of any exclusion, it means `include`

File: src/stripeJavaLanguageClient/utils.ts (hunk 2 above)
Comment (etsai-stripe) [reply src/stripeJavaLanguageClient/utils.ts:105]: i will update the output to include user next step

**Pull request merged**